minsar/utils/download_asfserial_rsmas.py

```diff
@@ -57,8 +57,29 @@ def main(iargs=None):
         os.remove(os.path.expanduser('~') + '/.bulk_download_cookiejar.txt')
     except OSError:
         pass
-
-    generate_files_csv(project_slc_dir, inps.custom_template_file)
+    
+    dataset_template = Template(inps.custom_template_file)
+    dataset_template.options.update(PathFind.correct_for_ssara_date_format(dataset_template.options))
```

Comment:
Can we do some commenting here about what the following code is actually doing? Preferably include things like: what the format of 'ssaraopt.startDate' looks like, what a 'seasonalDate' is, etc. 

Its a bit tough to understand what exactly is happening without provided context.

Reply:
```dataset_template = Template(inps.custom_template_file)
    dataset_template.options.update(PathFind.correct_for_ssara_date_format(dataset_template.options))
    subprocess.Popen("rm new_files.csv", shell=True).wait()
    standard_inputs = (inps, dataset_template)
    # this block provides seasonal date functionality: seasonal dates allow for the download of a specific period (less than a  year) multiple times in a year range
    if inps.seasonalStartDate is not None and inps.seasonalEndDate is not None:
        # a ssaraopt.startDate looks like this '12-07-2000'
        ogStartYearInt = int(dataset_template.options['ssaraopt.startDate'][:4])
```